### PR TITLE
Metrics Server addon.

### PR DESCRIPTION
**Pull request by camilb**
More info here:
https://kubernetes.io/docs/tasks/debug-application-cluster/core-metrics-pipeline/

Ref. #957

### CODE REVIEW

**Comment by danielfm** [comment]
What about heapster? Can this add-on run alongside heapster in such a way we can migrate to it in a safe manner?

**Comment by camilb** [comment]
Didn't have time to investigate it further, made a quick test in parallel with heapster in k8s 1.8 and works fine. `kubectl top` is one of the components that should use the new metrics-server but it didn't work without heapster.

**Comment by mumoshu** [comment]
@camilb Thanks for the contribution 👍 
This is conflicting with your latest work which has just been merged 😢 
Otherwise, LGTM.

**Comment by mumoshu** [comment]
@camilb Thanks for contributing!

File: core/controlplane/config/templates/cloud-config-controller [hunk 1]
```diff
@@ -2578,6 +2654,89 @@ write_files:
...
+          template:
+            metadata:
+              name: metrics-server
+              labels:
+                k8s-app: metrics-server
+            spec:
+              serviceAccountName: metrics-server
+              containers:
+              - name: metrics-server
+                image: gcr.io/google_containers/metrics-server-amd64:v0.2.0
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-controller]: shouldn't image be templated from  a value from cluster.yaml?

Comment (camilb) [reply core/controlplane/config/templates/cloud-config-controller]: yes, fixed in #1035

**Pull request merged**